add: Лазерный миниган РнД

Lentye opened the pull request:
<!-- Пишите **НИЖЕ** заголовков и **ВЫШЕ** комментариев, иначе ваш текст может не отобразиться. -->
<!-- В Contributing.MD вы можете найти некоторые рекомендации к оформлению пулл-реквеста. -->

## Описание
<!-- Опишите, что делает ваш ПР. Документировать каждую деталь не требуется, просто укажите основные изменения. -->
Добавление лазерного минигана в возможные творения РнД, арсенала Гамма шаттла и Гамма вендора

## Ссылка на предложение/Причина создания ПР
<!-- Здесь оставьте ссылку на сообщение в #отчеты-по-предложениям, чтобы подтвердить, что ваше предложение одобрено. Либо укажите, почему этот ПР должен пройти без предложки. -->
<!-- Пример ссылки: https://discord.com/channels/617003227182792704/755125334097133628/ID-сообщения -->
https://discord.com/channels/617003227182792704/755125334097133628/1224937140014547024

Gottfrei committed:
refactor: Runechat Tweaks (#5060)

actions-user committed:
Automatic changelog generation

Lentye committed:
Пустота внутри

Грустно, но мы тонем в дыме

Lentye committed:
fix

Lentye committed:
yeah

yeah

Gottfrei commented:
с ребейзом или мержем что-то не получилось я смотрю.

Lentye commented:
ну я тоже смотрю, вроде обещало без этих коммитов, брух

Lentye committed:
Revert "refactor: Runechat Tweaks (#5060)"

This reverts commit 2acc725a823b9e24acef9ac43f9798c866dbeec2.

	deleted:    code/__DEFINES/_helpers.dm
	modified:   code/__DEFINES/span.dm
	deleted:    code/__DEFINES/text.dm
	modified:   code/__HELPERS/type2type.dm
	modified:   code/__HELPERS/unsorted.dm
	modified:   code/datums/chatmessage.dm
	modified:   code/datums/emote/emote.dm
	modified:   code/defines/procs/records.dm
	modified:   code/game/atoms.dm
	modified:   code/game/atoms_movable.dm
	modified:   code/game/machinery/camera/tracking.dm
	modified:   code/game/machinery/computer/security.dm
	modified:   code/game/objects/items/devices/megaphone.dm
	modified:   code/modules/admin/verbs/possess.dm
	modified:   code/modules/client/client_procs.dm
	modified:   code/modules/mob/hear_say.dm
	modified:   code/modules/mob/living/carbon/human/human.dm
	modified:   code/modules/mob/living/carbon/human/life.dm
	modified:   code/modules/mob/living/living.dm
	modified:   code/modules/mob/living/living_say.dm
	modified:   code/modules/mob/living/silicon/ai/ai.dm
	modified:   code/modules/mob/living/silicon/silicon_say.dm
	modified:   code/modules/reagents/chemistry/reagents/misc.dm
	modified:   goon/browserassets/css/browserOutput-dark.css
	modified:   goon/browserassets/css/browserOutput.css
	modified:   paradise.dme

Lentye committed:
Revert "Automatic changelog generation"

This reverts commit bb4397494d65d38e25dfa85f0139ceaaeb63be6d.

	modified:   html/changelogs/archive/2024-06.yml

Lentye commented:
отревертил коммиты, жду респрайты от пипи и можно по сути вносить, как же давно я vi не щупал xd

Gottfrei commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -112,3 +112,63 @@
+		cell.maxcharge = 9000
+		cell.charge = 9000
+	fire_delay = 0.9
+	AddComponent(/datum/component/two_handed, \
+		force_unwielded = src.force_unwielded, \
+		force_wielded = src.force_wielded \
+	)
+	AddComponent(/datum/component/two_handed, require_twohands = TRUE)
```
зачем ты два раза подряд создаёшь один и тот же компонент? вероятно здесь должен быть /datum/component/automatic_fire

и вообще зачем тебе New()?

Lentye replied:
Потому что обожаю всратые идеи, на счёт автоматика никто не договаривался xd

Lentye replied:
А, ну да, Fire_delay округляется до 0 по умолчанию и я после создания должен поставить ему измененный вар ибо хочешь сделать фичу? Отрефактори 10 других xd

Gottfrei commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -112,3 +112,63 @@
+	..()
+	to_chat(user, span_warning("[name] слишком тяжелый!"))
+
+/obj/item/gun/energy/gun/minigun/update_icon_state()
+	..()
+	if(cell.charge > 600)
+		item_state = "gatling1"
+		icon_state = "gatling1"
+	else
+		item_state = "gatling"
+		icon_state = "gatling"
```
у энергетического оружия специфические обновления иконок, наследование здесь не нужно. плюс update_overlays заблокируй.

Lentye replied:
в ходе дебага без наследования было выяснено, что происходит рантайм инициализации ибо нет какой-то проверки на отсутствие предмета и случается RUNNING IN THE 90'S,IS A NEW WAY I LIKE TO BE

Gottfrei commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -112,3 +112,63 @@
+
+/obj/item/gun/energy/gun/minigun/Initialize(obj/item/gun/energy/gun/minigun/M)
+	..()
+	if(cell) //New() could not process gamma shuttle initialize,too late  cell appear xd
+		cell.maxcharge = 9000
+		cell.charge = 9000
```
сделай свою батарейку и присваивай через переменную cell_type

Gottfrei commented:
не та ветка: 
![84Q8sMU](https://github.com/ss220-space/Paradise/assets/35403274/ba53288c-dd38-44e6-bf42-40695e9d4bf1)

Lentye committed:
thunder

thunder

Lentye committed:
f

Lentye committed:
Вроде всё

только конфликт остался

Lentye committed:
experiment

experiment

Lentye commented:
Ну вроде всё что хотел сделать-сделал, если что-то надо еще, пингуйте в ДСе, я пошел заниматься своими делами

Lentye commented:
@Den109G  xddd

Gottfrei commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -112,3 +112,58 @@
+	var/force_unwielded = 10
+	var/force_wielded = 20
+
+/obj/item/gun/energy/gun/minigun/New(obj/item/gun/energy/gun/minigun/M)
```
использую Initialize

Lentye replied:
> использую Initialize
предлагаешь еще эти вары туда закинуть или чисто new заменить?

Lentye replied:
> использую Initialize

Ты предлагаешь оба вара туда запихнуть или чисто new заменить? странно отмечено xd

Gottfrei replied:
всё что тебе нужно делать в Initialize это вещать компонент.

Lentye replied:
не совсем, fire_delay очень специфический вар и игра стремиться его округлить с 0.9 до 0
![image](https://github.com/ss220-space/Paradise/assets/117032431/51f9a176-dd84-46c7-9c88-bec0bdd1897f)

Lentye replied:
его нельзя оставлять в варах и нужно ициировать отдельно

Gottfrei replied:
я не вижу ничего специфического в данной переменной. единственное место, где она меняется, это при обновлении типа патронов, во время инициализации оружия, следовательно тебе нужно оверрайдить именно этот прок.

правой кнопкой по переменной в VSCode -> Find All References

Lentye replied:
В конечном итоге так и оказалось, мм, в этом варианте я оставляю дэлэй в линзах xd

Gottfrei commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -112,3 +112,58 @@
+/obj/item/gun/energy/gun/minigun/examine(mob/user)
+	. = ..()
+
+	. += span_notice("Вы видите заряд батареи на [round(cell.charge/600)] залпов")
```
здесь нужен прок declension_ru для правильного склонения числа.

Lentye committed:
Доволен?

Lentye commented:
Довольна твоя душа?

Lentye committed:
here i am

dirty and faceless

Gottfrei commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -112,3 +112,56 @@
+	var/force_unwielded = 10
+	var/force_wielded = 20
+
+/obj/item/gun/energy/gun/minigun/Initialize(obj/item/gun/energy/gun/minigun/M)
+	..()
```
```suggestion
/obj/item/gun/energy/gun/minigun/Initialize(mapload)
	. = ..()
```

Gottfrei commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -112,3 +112,56 @@
+	..()
+	to_chat(user, span_warning("[name] слишком тяжелый!"))
+
+/obj/item/gun/energy/gun/minigun/update_icon_state()
+	if(cell.charge > 600)
```
```suggestion
/obj/item/gun/energy/gun/minigun/update_icon_state()
	if(!cell)
		item_state = initial(item_state)
		icon_state = initial(icon_state)
		return
	if(cell.charge > 600)
```

Gottfrei commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -112,3 +112,56 @@
+		item_state = "gatling"
+		icon_state = "gatling"
+
+/obj/item/gun/energy/gun/minigun/examine(mob/user)
+	. = ..()
+
+	. += span_notice("Вы видите заряд батареи на [round(cell.charge/600)] [declension_ru(cell.charge/600, "залп", "залпа", "залпов")]")
```
```suggestion
/obj/item/gun/energy/gun/minigun/examine(mob/user)
	. = ..()
	if(!cell)
		return .
	. += span_notice("Вы видите заряд батареи на [round(cell.charge/600)] [declension_ru(cell.charge/600, "залп", "залпа", "залпов")]")
```

Gottfrei committed:
Corrections

Gottfrei commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -112,3 +112,61 @@
+/obj/item/gun/energy/gun/minigun/examine(mob/user)
+	. = ..()
+	if(!cell)
+		return .
+	. += span_notice("Вы видите заряд батареи на [round(cell.charge/600)] [declension_ru(cell.charge/600, "залп", "залпа", "залпов")]")
```
```suggestion
		return .
	var/charge_amount = round(cell.charge/600)
	. += span_notice("Вы видите заряд батареи на [charge_amount] [declension_ru(charge_amount, "залп", "залпа", "залпов")]")
```

Gottfrei committed:
Double Calc

Lentye commented:
ну вот надо было мерджить что-то еще, теперь конфликт текстур xd

Lentye committed:
On branch laser-minigun
Your branch is up to date with 'origin/laser-minigun'.

Changes to be committed:
	modified:   icons/mob/inhands/guns_lefthand.dmi
	modified:   icons/mob/inhands/guns_righthand.dmi

Lentye committed:
On branch laser-minigun
Your branch is ahead of 'origin/laser-minigun' by 1 commit.
	(use "git push" to publish your local commits)

Changes to be committed:
	modified:   icons/obj/weapons/energy.dmi

Lentye committed:
asd

Lentye commented:
Лять, какой мердж конфликт, я ничего не удалял, я лишь добавлял, бл*

Lentye commented:
ну давай я удалю и снова запилю, гит сраный

Lentye committed:
On branch laser-minigun
Changes to be committed:
    modified:   icons/obj/weapons/energy.dmi

Den109G commented:
> ну давай я удалю и снова запилю

Лучше сделать ребейз, тогда будет понятно где проблема. Скорей всего в `dmi` файле, ибо там не как в текстовом документа сохраняются данные, а используется сжатие.

Lentye commented:
Почему это досих пор не закрыли...

Den109G commented:
Автор ПРа сам может закрыть ПР.
Если ты про то, почему не замержили - конфликты в ПРе.
